parser/src/parser/string.rs

```diff
@@ -7,37 +7,43 @@ use crate::token::Kind;
 use super::ast::Node;
 use crate::error::ParsingError;
 pub fn extract_string_inside(val: String) -> String {
-    if let Some(val) = val.strip_prefix("\"\"\"") {
-        val.strip_suffix("\"\"\"")
-            .expect("String must be enclosed with \"\"\"")
-            .to_string()
-    } else if let Some(val) = val.strip_prefix('\"') {
-        val.strip_suffix('\"')
-            .expect("String must be enclosed with \"")
-            .to_string()
-    } else if let Some(val) = val.strip_prefix("'''") {
-        val.strip_suffix("'''")
-            .expect("String must be enclosed with '''")
-            .to_string()
-    } else if let Some(val) = val.strip_prefix('\'') {
-        val.strip_suffix('\'')
-            .expect("String must be enclosed with '")
-            .to_string()
+ 
+
+    let delimiters = vec!["\"\"\"", "\"", "'''", "'"];
+    let mut result = String::new();
+    let is_raw = val.starts_with('r') || val.starts_with('R');
+    // drop first char if raw and put in val
+    let val = if is_raw {
+        val.chars().skip(1).collect::<String>()
     } else {
-        panic!(
-            "String must be enclosed in \"\"\", \"', ''' or ' but got {} ",
-            val.starts_with('\'')
-        );
+        val
+    };
+   
+    
+    for delimiter in delimiters {
+        if let Some(val) = val.strip_prefix(delimiter) {
+            let message = format!("String must be enclosed with {}", delimiter);
+            result = val
+                .strip_suffix(delimiter)
+                .expect(&message)
```

Comment:
Using expect here is an anti pattern because then we cannot show the error to user and program crashes. Although there is no chance of this error happening here since the lexer cannot tokenize the string if it does not have the ending quote still it's better to return errors here. We might improve lexer later to not crash on unterminated strings but to handle them somehow.

You can update this function signature to be able to return errors that are shown to the user like how we do it here:

https://github.com/Glyphack/enderpy/blob/be5402a36434585b924aa15339f5433a7325042b/parser/src/parser/parser.rs#L187